posthog/hogql/database/schema/util/where_clause_extractor.py

```diff
@@ -47,107 +41,168 @@ class SessionMinTimestampWhereClauseExtractor(CloningVisitor):
     context: HogQLContext
     clear_types: bool = False
     clear_locations: bool = False
+    capture_timestamp_comparisons: bool = False
+    our_tables: list[ast.LazyTable | ast.LazyJoin]
+    tombstone_string: str
+
+    limits: Literal["", "session_min"] = ""
 
-    def get_inner_where(self, parsed_query: ast.SelectQuery) -> Optional[ast.Expr]:
-        if not parsed_query.where:
+    def __init__(self, context: HogQLContext):
+        super().__init__()
+        self.context = context
+        self.our_tables = []
+        # A constant with this string will be used to escape early if we can't handle the query
+        self.tombstone_string = (
+            "__TOMBSTONE__" + ("".join(random.choices(string.ascii_uppercase + string.digits, k=10))) + "__"
+        )
```

Comment:
I'm unsure of the usage of this., especially considering no test cases are using this so the possible side effects of it may be unknown. Could you expand a bit more on why this is needed other than to escape early please?

Reply:
Chronologically speaking, the original approach that Robbie built, which I based this work on, returned just `true` or `false` constants from the different AST nodes. `true` meant "it's fine, ignore it", and `false` acted as the tombstone does now. Robbie only cared about capturing timestamp `CompareOperation`-s, but now I need to capture _and preserve_ all possible nested comparisons, including comparisons with boolean false-s, so I can't use the same system anymore. Thus magic tombstones.